modules/standard/Collection.chpl

```diff
@@ -0,0 +1,116 @@
+/*
+  A 'Collection' is a data structure, a container for elements that provide support
+  for insert, lookup, remove, and iteration operations.
+*/
+class Collection {
+  /*
+    The type of element that this Collection holds.
+  */
+  type eltType;
+
+  /*
+    Adds an element to this data structure.
+  */
+  inline proc add(elt : eltType) : bool {
+    halt("'proc add(elt : eltType) : bool' is not supported...");
+  }
+
+  /*
+    Removes an arbitrary element from this data structure.
+
+    **BUG:** Compiler will segfault if the returned value is not captured at callsite.
+    Issue: #6542
+
+    **FIX:** Ensure that you always capture the return value...
+
+    ::
+
+      var capturedRetval = c.remove()
+
+    **BUG:** Loop Invariant Code Motion causes undefined behavior if assigned to a
+    variable declared outside of loop. Issue: #7003
+
+    **FIX:** Use the `--no-loop-invariant-code-motion` to disable LICM.
+    Otherwise, just make sure you always capture the return value inside of a loop
+    in a variable not declared outside of loop...
+
+    ::
+
+      for i in 1 .. N {
+        var retval = c.remove();
+      }
+
+  */
+  inline proc remove() : (bool, eltType) {

```

Comment:
Right, I didn't want to imply any particular ordering, and `add` and `remove` seemed descriptive enough, as `remove` in this case means "remove *any* item".